### ISSUE TITLE
The rescan_library function is crashing SAM

### ISSUE THREAD

**Issue by diegolinhares** [NREL/SAM#544]
**Describe the bug**
I was using a script to add a folder to the solar library. When I run the script, SAM crashes.

I think it's some memory/access problem with *rescan_library*.

**To Reproduce**

Use this script:

```
create_case('PVWatts', 'Commercial', 'PVWattsCommercial');

downloaded_solar_files_path = homedir() + '/solar_files/';
set_settings('solar_data_paths', downloaded_solar_files_path);
rescanlibrary('solar');
```

**Expected behavior**
Add my folder to SAM's library.

**Comment by cpaulgilman** [comment]
@diegolinhares Thank you for reporting this bug and including the LK sample. I was able to replicate the problem and will investigate further. Are you having any trouble adding a weather file folder from the Solar Resource input page, or does this only happen from an LK script?

Thanks,
Paul.

**Comment by diegolinhares** [comment]
> @diegolinhares Thank you for reporting this bug and including the LK sample. I was able to replicate the problem and will investigate further. Are you having any trouble adding a weather file folder from the Solar Resource input page, or does this only happen from an LK script?
> 
> Thanks,
> Paul.

Hey Paul,

The problem is only when adding a folder from an LK script.

Thanks,
Diego

**Comment by sjanzou** [comment]
fixed with pull request #611
